sql/core/src/main/scala/org/apache/spark/sql/execution/exchange.scala

```diff
@@ -261,10 +266,71 @@ object Exchange {
 }
 
 /**
+ * A [[BroadcastExchange]] collects, transforms and finally broadcasts the result of a transformed
+ * SparkPlan.
+ */
+case class BroadcastExchange(
```

Comment:
i think it's a good chance to create the "exchange" package and move everything in this file there as multiple files. we can do that later though.